src/AppInstallerCommonCore/Authentication/WebAccountManagerAuthenticator.cpp

```diff
@@ -0,0 +1,287 @@
+            AICLI_LOG(Core, Error, << "CreateTokenRequest returned empty request");
+            return {};
+        }
+
+        IAsyncOperation<WebTokenRequestResult> requestOperation;
+        constexpr winrt::guid iidAsyncRequestResult{ winrt::guid_of<IAsyncOperation<WebTokenRequestResult>>() };
+        auto authManagerFactory = winrt::get_activation_factory<WebAuthenticationCoreManager>();
+        winrt::com_ptr<IWebAuthenticationCoreManagerInterop> authManagerInterop{ authManagerFactory.as<IWebAuthenticationCoreManagerInterop>() };
+
+        HRESULT requestOperationResult = APPINSTALLER_CLI_ERROR_AUTHENTICATION_FAILED;
+        AuthenticationWindowBase parentWindow;
+        if (webAccount)
+        {
+            requestOperationResult = authManagerInterop->RequestTokenWithWebAccountForWindowAsync(
+                parentWindow.GetHandle(),
+                request.as<::IInspectable>().get(),
+                webAccount.as<::IInspectable>().get(),
+                iidAsyncRequestResult,
+                reinterpret_cast<void**>(&requestOperation));
+        }
+        else
+        {
+            requestOperationResult = authManagerInterop->RequestTokenForWindowAsync(
+                parentWindow.GetHandle(),
+                request.as<::IInspectable>().get(),
+                iidAsyncRequestResult,
+                reinterpret_cast<void**>(&requestOperation));
+        }
+
+        if (FAILED(requestOperationResult))
+        {
+            AICLI_LOG(Core, Error, << "RequestTokenForWindowAsync failed. Result: " << requestOperationResult);
+            return {};
+        }
+
+        return HandleGetTokenResult(requestOperation.get());
+    }
+
+    AuthenticationResult WebAccountManagerAuthenticator::GetTokenSilent(WebAccount webAccount)
+    {
+        AICLI_LOG(Core, Info, << "Started GetTokenSilent.");
+
+        auto request = CreateTokenRequest(false);
+        if (!request)
+        {
+            AICLI_LOG(Core, Error, << "CreateTokenRequest returned empty request");
+            return {};
+        }
+
+        return HandleGetTokenResult(WebAuthenticationCoreManager::GetTokenSilentlyAsync(request).get());
+    }
+
+    AuthenticationResult WebAccountManagerAuthenticator::HandleGetTokenResult(WebTokenRequestResult requestResult)
+    {
+        AuthenticationResult result;
+
+        if (!requestResult)
+        {
+            AICLI_LOG(Core, Error, << "WebTokenRequestResult is null");
+            return result;
+        }
+
+        if (requestResult.ResponseStatus() == WebTokenRequestStatus::Success)
+        {
+            auto responseData = requestResult.ResponseData().GetAt(0);
+            auto authenticatedAccount = responseData.WebAccount();
+
+            // Check token's corresponding account matches user input if applicable.
+            if (m_authArgs.AuthenticationAccount.empty() || Utility::CaseInsensitiveEquals(m_authArgs.AuthenticationAccount, Utility::ConvertToUTF8(authenticatedAccount.UserName())))
```

Comment:
>m_authArgs.AuthenticationAccount.empty() [](http://example.com/codeflow?start=16&length=40)

Does an empty authenticationAccount mean that it succeeded?

Reply:
Empty m_authArgs.AuthenticationAccount means user or caller did not specify which account to use. So we consider it as authentication success as long as WebAccountmanager returns success.